CLI List and HCP command docs refresh (#28778)
hashicorp/vault: website/content/docs/commands/list.mdx

---
layout: docs
page_title: list - Vault CLI
description: |-
  List data from Vault at the given path. This can be used
  to list keys in a, given secrets engine.
---

# list

> List data from Vault at the given path. This can be used
> to list keys in a, given secrets engine.

Reviewer: ```suggestion
List arbitrary data from Vault at the given path. For example, list the keys
stored under a specific mount path for a secrets plugin.
```

<CodeBlockConfig hideClipboard>

```shell-session

> $ vault list [options] PATH

Reviewer: ```suggestion
$ vault list [flags] <vault_path>
```

It doesn't look like the command supports any options, just the one argument and standard flags

$ vault list [-help | -h]
```

</CodeBlockConfig>

> ## Description

Reviewer: Is there a corresponding API endpoint?

`list` lists data from Vault at the given path (wrapper command for
HTTP LIST). This can be used to list keys in a given secrets engine.

> ### Limitations and warnings
> 
> - None

Reviewer: ```suggestion
```

## Command arguments

<br />

@include 'cli/list/args/path.mdx'

> ## Command flags

Reviewer: ```suggestion
## Command options

None.

## Command flags
```

> - None

Reviewer: ```suggestion
None.
```
